WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -832,16 +836,26 @@ private void SetFeeTargetLimits()
 			}
 		}
 
-		private bool TryParseUserFee(out decimal userFee)
+		public bool TryParseUserFee(out decimal userFeeRateAsDecimal) => TryParseUserFeeCore(UserFeeText, out userFeeRateAsDecimal);
+
+		public static bool TryParseUserFeeCore(string feeText, out decimal userFeeRateAsDecimal)
 		{
-			userFee = default;
-			var userFeeText = UserFeeText;
-			return
-				userFeeText is { }
-				&& !userFeeText.Contains(",")
-				&& decimal.TryParse(userFeeText, out userFee)
-				&& (userFee * 1_000) < Constants.MaxSatoshisSupply
-				&& userFee > 0;
+			try
+			{
+				if (decimal.TryParse(feeText, NumberStyles.Integer, new CultureInfo("en-US"), out var userFee))
```

Comment:
I think you should use `NumberStyles.AllowDecimalPoint` but one edge case I see with this is that `0.99999999999999999999999999999` will be considered as valid. I think having max length property for controls is necessary but not for this PR.